src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -38,6 +39,13 @@
     public static final String PROCESS_NAME_REGEX = "process_name_regex";
     public static final String ATTRIBUTE = "Attribute: ";
 
+    private static final ThreadLocal<Yaml> YAML = new ThreadLocal<Yaml>() {
```

Comment:
Why does it need to be a `ThreadLocal` object?

Reply:
There seemed to be non-trivial amount of initialization happening in the constructor, so I wanted to avoid it.  The problem is (I believe) `Yaml` is not thread safe, so I used the ThreadLocal to ensure that safety.

Reply:
ok I trust you judgement here